avro-builder/src/main/java/com/linkedin/avroutil1/builder/operations/codegen/own/AvroUtilCodeGenOp.java

```diff
@@ -46,38 +47,38 @@ public AvroUtilCodeGenOp(CodeGenOpConfig config) {
 
   @Override
   public void run(OperationContext opContext) throws Exception {
-    //mkdir any output folders that dont exist
+    //mkdir any output folders that don't exist
     if (!config.getOutputSpecificRecordClassesRoot().exists() && !config.getOutputSpecificRecordClassesRoot()
         .mkdirs()) {
       throw new IllegalStateException(
           "unable to create destination folder " + config.getOutputSpecificRecordClassesRoot());
     }
 
-    List<Path> avscFiles = new ArrayList<>();
+    Set<File> avscFiles = new HashSet<>();
+    Set<File> nonImportableFiles = new HashSet<>();
+    String[] extensions = new String[]{BuilderConsts.AVSC_EXTENSION};
+
     for (File inputRoot : config.getInputRoots()) {
-      Files.walk(inputRoot.toPath())
-          .filter(path -> path.getFileName().toString().endsWith("." + BuilderConsts.AVSC_EXTENSION))
-          .forEach(avscFiles::add);
+      avscFiles.addAll(FileUtils.listFiles(inputRoot, extensions, true));
+    }
+
+    if (config.getNonImportableSourceRoots() != null) {
+      for (File nonImportableRoot : config.getNonImportableSourceRoots()) {
+        nonImportableFiles.addAll(FileUtils.listFiles(nonImportableRoot, extensions, true));
+      }
     }
-    if (avscFiles.isEmpty()) {
+
+    if (avscFiles.isEmpty() && nonImportableFiles.isEmpty()) {
       LOGGER.warn("no input schema files were found under roots " + config.getInputRoots());
       return;
     }
-    LOGGER.info("found " + avscFiles.size() + " avsc schema files");
+    LOGGER.info("found " + (avscFiles.size() + nonImportableFiles.size()) + " avsc schema files");
 
     AvroParseContext context = new AvroParseContext();
-    AvscParser parser = new AvscParser();
     List<AvscParseResult> parsedFiles = new ArrayList<>();
 
-    for (Path p : avscFiles) {
-      AvscParseResult fileParseResult = parser.parse(p);
-      Throwable parseError = fileParseResult.getParseError();
-      if (parseError != null) {
-        throw new IllegalArgumentException("failed to parse file " + p.toAbsolutePath(), parseError);
-      }
-      context.add(fileParseResult);
-      parsedFiles.add(fileParseResult);
-    }
+    parseAvscFiles(avscFiles, true, context, parsedFiles);
+    parseAvscFiles(nonImportableFiles, false, context, parsedFiles);
```

Comment:
now that we have multiple "parse runs" - do we need to check for duplicate definitions across them ? (adding a TODO would suffice for now)

Reply:
Good point. There's a TODO in this file still so I'll keep it there for now.